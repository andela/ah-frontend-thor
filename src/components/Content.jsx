import React from "react";

const Content = () => {
	return (
		<div className="container">
			<form>
				<div className="form-group">
					<input
						type="email"
						className="form-control"
						placeholder="Enter email"
					/>
				</div>
				<div className="form-group">
					<input
						type="password"
						className="form-control"
						placeholder="Enter Password"
					/>
				</div>
				<button type="submit" className="btn btn-primary">
					Login
				</button>
			</form>
		</div>
	);
};
export default Content;
