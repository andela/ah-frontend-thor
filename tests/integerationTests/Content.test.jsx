import { shallow } from "enzyme";
import React from "react";
import Content from "../../src/components/Content";

describe("Tests the mounted Footer component", () => {
  it("tests the footer of the app against the snapshot", () => {
    const component = shallow(<Content />);
    expect(component).toMatchSnapshot();
  });
});
