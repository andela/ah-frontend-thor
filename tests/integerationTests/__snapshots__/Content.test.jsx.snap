// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests the mounted Footer component tests the footer of the app against the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Content />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <form>
        <div
          className="form-group"
        >
          <input
            className="form-control"
            placeholder="Enter email"
            type="email"
          />
        </div>
        <div
          className="form-group"
        >
          <input
            className="form-control"
            placeholder="Enter Password"
            type="password"
          />
        </div>
        <button
          className="btn btn-primary"
          type="submit"
        >
          Login
        </button>
      </form>,
      "className": "container",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="form-group"
          >
            <input
              className="form-control"
              placeholder="Enter email"
              type="email"
            />
          </div>,
          <div
            className="form-group"
          >
            <input
              className="form-control"
              placeholder="Enter Password"
              type="password"
            />
          </div>,
          <button
            className="btn btn-primary"
            type="submit"
          >
            Login
          </button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <input
              className="form-control"
              placeholder="Enter email"
              type="email"
            />,
            "className": "form-group",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "form-control",
              "placeholder": "Enter email",
              "type": "email",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <input
              className="form-control"
              placeholder="Enter Password"
              type="password"
            />,
            "className": "form-group",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "form-control",
              "placeholder": "Enter Password",
              "type": "password",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Login",
            "className": "btn btn-primary",
            "type": "submit",
          },
          "ref": null,
          "rendered": "Login",
          "type": "button",
        },
      ],
      "type": "form",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <form>
          <div
            className="form-group"
          >
            <input
              className="form-control"
              placeholder="Enter email"
              type="email"
            />
          </div>
          <div
            className="form-group"
          >
            <input
              className="form-control"
              placeholder="Enter Password"
              type="password"
            />
          </div>
          <button
            className="btn btn-primary"
            type="submit"
          >
            Login
          </button>
        </form>,
        "className": "container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="form-group"
            >
              <input
                className="form-control"
                placeholder="Enter email"
                type="email"
              />
            </div>,
            <div
              className="form-group"
            >
              <input
                className="form-control"
                placeholder="Enter Password"
                type="password"
              />
            </div>,
            <button
              className="btn btn-primary"
              type="submit"
            >
              Login
            </button>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <input
                className="form-control"
                placeholder="Enter email"
                type="email"
              />,
              "className": "form-group",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "form-control",
                "placeholder": "Enter email",
                "type": "email",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <input
                className="form-control"
                placeholder="Enter Password"
                type="password"
              />,
              "className": "form-group",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "form-control",
                "placeholder": "Enter Password",
                "type": "password",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Login",
              "className": "btn btn-primary",
              "type": "submit",
            },
            "ref": null,
            "rendered": "Login",
            "type": "button",
          },
        ],
        "type": "form",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
